app/services/google/parser_service.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module Google
+  class ParserService
+    TOP_ADS_SECTION_ID = 'tvcap'
+    BOTTOM_ADS_SECTION_ID = 'bottomads'
+    ADS_CLASS = 'uEierd'
+    SEARCH_SECTION_ID = 'search'
+    SEARCH_RESULT_CLASS = 'yuRUbf'
+
+    def initialize(html)
+      @html = html
+      @doc = Nokogiri::HTML(html)
+    end
+
+    def call
+      {
+        ads_top_count: ads_top_count,
+        ads_page_count: ads_page_count,
+        ads_top_urls: ads_top_urls,
+        result_count: result_count,
+        result_urls: result_urls,
+        total_link_count: total_link_count,
+        html: html
+      }
+    end
+
+    private
+
+    attr_reader :html, :doc
+
+    def ads_top_count
+      doc.css("div[id='#{TOP_ADS_SECTION_ID}']").css("div[class='#{ADS_CLASS}']").count
+    end
+
+    def ads_page_count
+      doc.css("div[class='#{ADS_CLASS}']").count
```

Comment:
```suggestion
      @ads_page_count ||= doc.css("div[class='#{ADS_CLASS}']").count
```

As we are calling this method twice, it is interesting to cache the value.

Same for the `result_count` method ;-)